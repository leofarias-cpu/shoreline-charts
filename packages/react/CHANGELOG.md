# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.1.0-labs.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.1...@vtex/onda-react@0.1.0-labs.2) (2021-05-13)


### Features

* **onda-react:** allow state property ([47bfb26](https://github.com/vtex/onda/commit/47bfb26bdb2f2f5496a9458505913f8133dfbee2))





# [0.1.0-labs.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.0...@vtex/onda-react@0.1.0-labs.1) (2021-05-13)

**Note:** Version bump only for package @vtex/onda-react





# 0.1.0-labs.0 (2021-05-13)


### Features

* **onda:** add /react package ([4728b63](https://github.com/vtex/onda/commit/4728b631011945c4bd25a6e2de57c2d2e4f52c17))
