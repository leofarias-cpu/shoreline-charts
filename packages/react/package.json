{
  "name": "@vtex/admin-ui-react",
  "license": "MIT",
  "version": "0.0.0",
  "main": "dist/index.js",
  "module": "dist/admin-ui-react.esm.js",
  "typings": "dist/index.d.ts",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist",
    "src"
  ],
  "engines": {
    "node": ">=10"
  },
  "repository": {
    "directory": "packages/react",
    "type": "git",
    "url": "git+https://github.com/vtex/onda.git"
  },
  "bugs": {
    "url": "https://github.com/vtex/onda/issues"
  },
  "scripts": {
    "start": "tsdx watch --verbose --noClean",
    "build": "tsdx build --tsconfig tsconfig.json",
    "build:fast": "tsdx build --tsconfig tsconfig.json --transpileOnly --format esm,cjs",
    "test": "tsdx test"
  },
  "peerDependencies": {
    "@emotion/css": ">=11",
    "@emotion/react": ">=11",
    "react": ">=16",
    "react-dom": ">=16"
  },
  "devDependencies": {
    "@vtex/onda-core": "^0.1.0-labs.8",
    "polished": "^4.1.2"
  },
  "dependencies": {
    "@emotion/is-prop-valid": "^1.1.0",
    "@vtex/onda-util": "^0.1.0-labs.3",
    "react-is": "^17.0.2",
    "tiny-invariant": "^1.1.0",
    "tiny-warning": "^1.0.3"
  }
}
