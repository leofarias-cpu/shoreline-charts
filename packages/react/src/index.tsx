export * from './tag'
export * from './domElements'
export * from './jsx'
export * from './types'
export * from './useStylesheet'
export * from './util'
export * from './symbols'
export * from './hooks/useMediaQuery'
export * from './hooks/useResponsiveValue'
