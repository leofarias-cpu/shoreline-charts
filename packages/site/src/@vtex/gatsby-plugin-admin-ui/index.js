export default {
  colors: {
    purple: '#805ad5',
    rebelPink: '#F71963'
  },
  sizes: {
    header: '4.5rem',
    content: '90rem',
  },
  space: {
    content: '4.5rem',
  },
}
