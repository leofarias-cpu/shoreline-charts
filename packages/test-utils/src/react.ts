import {
  render,
  renderHook,
  act,
  screen,
  fireEvent,
} from '@testing-library/react'

export { render, renderHook, act, screen, fireEvent }
