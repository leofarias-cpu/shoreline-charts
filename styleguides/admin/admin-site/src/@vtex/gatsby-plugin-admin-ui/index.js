export default {
  sizes: {
    header: '4.5rem',
    content: '90rem',
  },
  space: {
    content: '4.5rem',
  },
}
