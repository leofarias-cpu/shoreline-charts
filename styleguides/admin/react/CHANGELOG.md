# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.4.3](https://github.com/vtex/onda/compare/@vtex/onda-react@0.4.2...@vtex/onda-react@0.4.3) (2021-07-20)

**Note:** Version bump only for package @vtex/onda-react





## [0.4.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.4.1...@vtex/onda-react@0.4.2) (2021-07-19)

**Note:** Version bump only for package @vtex/onda-react





## [0.4.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.4.0...@vtex/onda-react@0.4.1) (2021-07-16)


### Bug Fixes

* **onda-react:** remove the cleaning of options properties from jsx ([b8ec31c](https://github.com/vtex/onda/commit/b8ec31cf1e557996ea8eb7027edbcfcbe55845fb))





# [0.4.0](https://github.com/vtex/onda/compare/@vtex/onda-react@0.3.0...@vtex/onda-react@0.4.0) (2021-07-15)


### Features

* **admin-ui/ultimate-table:** support sorting ([7453750](https://github.com/vtex/onda/commit/745375001bd28bb73e8a229f14c12e78bd0385da))





# [0.3.0](https://github.com/vtex/onda/compare/@vtex/onda-react@0.2.2...@vtex/onda-react@0.3.0) (2021-07-15)


### Bug Fixes

* **onda-react:** jsx refs type inference ([5a24609](https://github.com/vtex/onda/commit/5a2460960c419a272f16d76a43f9575247643f2a))
* **onda-react:** jsx rendering issues ([6fbe2b5](https://github.com/vtex/onda/commit/6fbe2b585ab083a99a1aa806de3ed0e23ac2451f))
* **onda-react:** typescript inference for jsx.elements ([fb9e365](https://github.com/vtex/onda/commit/fb9e365df6a877a274bc3ea26131a4015fbbcd33))


### Features

* **onda-react:** add ComponentProps type ([e13433c](https://github.com/vtex/onda/commit/e13433c2b9611de3cd6723541fe82064bd1f10cb))





## [0.2.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.2.1...@vtex/onda-react@0.2.2) (2021-07-05)

**Note:** Version bump only for package @vtex/onda-react





## [0.2.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.2.0...@vtex/onda-react@0.2.1) (2021-07-01)


### Bug Fixes

* **onda-core:** build imports ([9d30f9e](https://github.com/vtex/onda/commit/9d30f9e743d75b03995547e4d3052aa5a510e095))





# [0.2.0](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.1...@vtex/onda-react@0.2.0) (2021-07-01)


### Bug Fixes

* **onda-react:** remove unnecessary state ([1465e69](https://github.com/vtex/onda/commit/1465e695a6cb9be6ba3773d0acf51194a0713ea2))
* **onda-react:** resolve jsx hof types ([2e185ac](https://github.com/vtex/onda/commit/2e185ac2dd319f30d1eb4332a13a2299a38e06cb))


### Features

* **onda-react:** rephrase createComponent as jsx ([dd157ff](https://github.com/vtex/onda/commit/dd157ffa2500ed8e911b7d2d04db65268fc7f27b))





## [0.1.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.4...@vtex/onda-react@0.1.1) (2021-06-29)


### Bug Fixes

* **onda-react:** package.json version ([fb98ac4](https://github.com/vtex/onda/commit/fb98ac411f80ae253225940a7c687cb458677e4a))





# 0.1.0-labs.4 (2021-06-24)


### Features

* **onda-packages:** add onda packages ([b619064](https://github.com/vtex/onda/commit/b619064d04d190d5615a2832dbcb86d96efd16ee))





# [0.1.0-labs.3](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.2...@vtex/onda-react@0.1.0-labs.3) (2021-05-19)


### Features

* **onda:** allow global styles from theme ([aff3fa4](https://github.com/vtex/onda/commit/aff3fa449c6b92004312bf8ee4f648d7e6258dc0))
* **onda-core:** transform theme values on css variables ([3a8a379](https://github.com/vtex/onda/commit/3a8a379c3ccebf4738b65d39fc38aa8ac594e621))
* **onda-runtime-emotion:** allow default entries ([73a0dea](https://github.com/vtex/onda/commit/73a0deafda0b18ca09ca6eca00637d0c976055d8))





# [0.1.0-labs.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.1...@vtex/onda-react@0.1.0-labs.2) (2021-05-13)


### Features

* **onda-react:** allow state property ([47bfb26](https://github.com/vtex/onda/commit/47bfb26bdb2f2f5496a9458505913f8133dfbee2))





# [0.1.0-labs.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.0...@vtex/onda-react@0.1.0-labs.1) (2021-05-13)

**Note:** Version bump only for package @vtex/onda-react





# 0.1.0-labs.0 (2021-05-13)


### Features

* **onda:** add /react package ([4728b63](https://github.com/vtex/onda/commit/4728b631011945c4bd25a6e2de57c2d2e4f52c17))
