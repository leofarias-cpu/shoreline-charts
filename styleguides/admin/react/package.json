{
  "name": "@vtex/onda-react",
  "license": "MIT",
  "version": "0.1.0",
  "main": "dist/index.js",
  "module": "dist/onda-react.esm.js",
  "typings": "dist/index.d.ts",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist",
    "src"
  ],
  "engines": {
    "node": ">=10"
  },
  "repository": {
    "directory": "packages/react",
    "type": "git",
    "url": "git+https://github.com/vtex/onda.git"
  },
  "bugs": {
    "url": "https://github.com/vtex/onda/issues"
  },
  "scripts": {
    "start": "tsdx watch --verbose --noClean",
    "build": "tsdx build --tsconfig tsconfig.json",
    "build:fast": "tsdx build --tsconfig tsconfig.json --transpileOnly --format esm,cjs",
    "test": "tsdx test --passWithNoTests",
    "storybook": "start-storybook -p 6005",
    "storybook:build": "yarn --cwd ../../ build && yarn run build-storybook"
  },
  "peerDependencies": {
    "@emotion/css": ">=11",
    "@emotion/react": ">=11",
    "react": ">=16",
    "react-dom": ">=16"
  },
  "devDependencies": {
    "@vtex/admin-core": "^0.10.10",
    "@vtex/onda-util": "^0.1.0-labs.2",
    "polished": "^4.1.2"
  },
  "dependencies": {
    "@emotion/is-prop-valid": "^1.1.0",
    "react-is": "^17.0.2",
    "tiny-invariant": "^1.1.0",
    "tiny-warning": "^1.0.3"
  }
}
