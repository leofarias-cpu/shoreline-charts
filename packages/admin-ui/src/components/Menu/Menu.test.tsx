import React from 'react'
import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import 'mutationobserver-shim'

import { ThemeProvider } from '@vtex/admin-core'

import { Menu } from './index'

global.MutationObserver = window.MutationObserver

describe('Menu tests', () => {
  it('should have overridable styles', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <Menu
          visible
          data-testid="menu"
          data-boxtestid="menubox"
          aria-label="menu label"
          csx={{
            bg: 'coral',
          }}
          disclosure={<button>Open</button>}
        >
          <Menu.Item>Download</Menu.Item>
          <Menu.Item>Link to</Menu.Item>
          <Menu.Item>Favorite</Menu.Item>
          <Menu.Separator />
          <Menu.Item>Delete</Menu.Item>
        </Menu>
      </ThemeProvider>
    )

    expect(getByTestId('menubox')).toHaveStyleRule('background-color', 'coral')
  })

  it('should match snapshot visible', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Menu
          visible
          data-testid="menu"
          data-boxtestid="menubox"
          aria-label="menu label"
          disclosure={<button>Open</button>}
          baseId="id"
        >
          <Menu.Item>Download</Menu.Item>
          <Menu.Item>Link to</Menu.Item>
          <Menu.Item>Favorite</Menu.Item>
          <Menu.Separator />
          <Menu.Item>Delete</Menu.Item>
        </Menu>
      </ThemeProvider>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot hidden', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Menu
          data-testid="menu"
          data-boxtestid="menubox"
          aria-label="menu label"
          disclosure={<button>Open</button>}
          baseId="id"
        >
          <Menu.Item>Download</Menu.Item>
          <Menu.Item>Link to</Menu.Item>
          <Menu.Item>Favorite</Menu.Item>
          <Menu.Separator />
          <Menu.Item>Delete</Menu.Item>
        </Menu>
      </ThemeProvider>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should not have a11y violations', async () => {
    const { container } = render(
      <ThemeProvider>
        <Menu
          visible
          data-testid="menu"
          data-boxtestid="menubox"
          aria-label="menu label"
          disclosure={<button>Open</button>}
        >
          <Menu.Item>Download</Menu.Item>
          <Menu.Item>Link to</Menu.Item>
          <Menu.Item>Favorite</Menu.Item>
          <Menu.Item>Delete</Menu.Item>
        </Menu>
      </ThemeProvider>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
