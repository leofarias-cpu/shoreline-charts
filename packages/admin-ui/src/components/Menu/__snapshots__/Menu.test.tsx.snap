// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match snapshot hidden 1`] = `
<DocumentFragment>
  

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: #FFFFFF;
  background: #2953B2;
  padding: 0.5rem 0.75rem;
  height: 2.25rem;
}

.emotion-3:hover {
  color: #FFFFFF;
  background: #20408C;
}

.emotion-3:active {
  color: #FFFFFF;
  background: #172F66;
}

.emotion-3:disabled {
  background: #E6E6E6;
  color: #747474;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F1F4FC,0rem 0rem 0rem 0.1875rem #96B2F2;
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  visibility: visible;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-1 svg {
  margin-left: 0.125rem;
}

.emotion-0 {
  height: 1rem;
  width: 1rem;
  min-height: 1rem;
  min-width: 1rem;
}

.emotion-13 {
  outline: none;
  z-index: 999;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: coral;
  padding: 0.75rem;
  min-width: 18px;
  border: 1px solid #DDDDDD;
  border-radius: 3px;
  box-shadow: 0rem 0rem 1rem 0rem rgba(0,0,0,0.10);
}

.emotion-5 {
  margin-top: 0.125rem;
  margin-bottom: 0.125rem;
  padding-left: 0.25rem;
  padding-right: 0.25rem;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  border: none;
  text-transform: initial;
  width: 100%;
  height: 32px;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  background: transparent;
  color: #2953B2;
}

.emotion-5 div {
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-5 svg {
  margin: 0.25rem;
  width: 20px;
  height: 20px;
  min-width: 20px;
  min-height: 20px;
  margin-left: 0rem;
  margin-right: 0.5rem;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F1F4FC,0rem 0rem 0rem 0.1875rem #96B2F2;
}

.emotion-5:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-5:hover {
  color: #20408C;
  background: #F1F4FC;
}

.emotion-5:active {
  color: #172F66;
  background: #DDE5F7;
}

.emotion-5:disabled {
  color: #747474;
  background: #E6E6E6;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-10 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  border-bottom: 1px solid #DDDDDD;
  width: calc(100% -0.75rem);
  margin-left: -0.75rem;
  margin-right: -0.75rem;
  outline: none;
}

<div
    class="emotion-14"
  >
    <button
      aria-controls="id"
      aria-expanded="false"
      aria-haspopup="menu"
      class="emotion-3"
      type="button"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-1"
        >
          <svg
            class="emotion-0"
            fill="currentColor"
            height="1rem"
            viewBox="0 0 256 256"
            width="1rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="16rem"
              width="16rem"
            />
            <polyline
              fill="none"
              points="208 96 128 176 48 96"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="18"
            />
          </svg>
          Open
        </div>
      </div>
    </button>
    <div
      aria-label="menu label"
      aria-orientation="vertical"
      class="emotion-13"
      data-dialog="true"
      data-testid="menu"
      hidden=""
      id="id"
      role="menu"
      style="display: none; position: fixed; left: 100%; top: 100%;"
      tabindex="-1"
    >
      <button
        class="emotion-5"
        id="id-1"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        <div
          class="emotion-4"
        >
           Download
        </div>
      </button>
      <button
        class="emotion-5"
        id="id-2"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        <div
          class="emotion-4"
        >
           Link to
        </div>
      </button>
      <button
        class="emotion-5"
        id="id-3"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        <div
          class="emotion-4"
        >
           Favorite
        </div>
      </button>
      <hr
        aria-orientation="horizontal"
        class="emotion-10"
        role="separator"
      />
      <button
        class="emotion-5"
        id="id-4"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        <div
          class="emotion-4"
        >
           Delete
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Menu should match snapshot visible 1`] = `
<DocumentFragment>
  

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: #FFFFFF;
  background: #2953B2;
  padding: 0.5rem 0.75rem;
  height: 2.25rem;
}

.emotion-3:hover {
  color: #FFFFFF;
  background: #20408C;
}

.emotion-3:active {
  color: #FFFFFF;
  background: #172F66;
}

.emotion-3:disabled {
  background: #E6E6E6;
  color: #747474;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F1F4FC,0rem 0rem 0rem 0.1875rem #96B2F2;
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  visibility: visible;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-1 svg {
  margin-left: 0.125rem;
}

.emotion-0 {
  height: 1rem;
  width: 1rem;
  min-height: 1rem;
  min-width: 1rem;
}

.emotion-5 {
  margin-top: 0.125rem;
  margin-bottom: 0.125rem;
  padding-left: 0.25rem;
  padding-right: 0.25rem;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  border: none;
  text-transform: initial;
  width: 100%;
  height: 32px;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  background: transparent;
  color: #2953B2;
}

.emotion-5 div {
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-5 svg {
  margin: 0.25rem;
  width: 20px;
  height: 20px;
  min-width: 20px;
  min-height: 20px;
  margin-left: 0rem;
  margin-right: 0.5rem;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F1F4FC,0rem 0rem 0rem 0.1875rem #96B2F2;
}

.emotion-5:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-5:hover {
  color: #20408C;
  background: #F1F4FC;
}

.emotion-5:active {
  color: #172F66;
  background: #DDE5F7;
}

.emotion-5:disabled {
  color: #747474;
  background: #E6E6E6;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-10 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  border-bottom: 1px solid #DDDDDD;
  width: calc(100% -0.75rem);
  margin-left: -0.75rem;
  margin-right: -0.75rem;
  outline: none;
}

.emotion-13 {
  outline: none;
  z-index: 999;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #FFFFFF;
  padding: 0.75rem;
  min-width: 18px;
  border: 1px solid #DDDDDD;
  border-radius: 3px;
  box-shadow: 0rem 0rem 1rem 0rem rgba(0,0,0,0.10);
}

<div
    class="emotion-14"
  >
    <button
      aria-controls="id"
      aria-expanded="true"
      aria-haspopup="menu"
      class="emotion-3"
      type="button"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-1"
        >
          <svg
            class="emotion-0"
            fill="currentColor"
            height="1rem"
            viewBox="0 0 256 256"
            width="1rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="16rem"
              width="16rem"
            />
            <polyline
              fill="none"
              points="208 96 128 176 48 96"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="18"
            />
          </svg>
          Open
        </div>
      </div>
    </button>
    <div
      aria-label="menu label"
      aria-orientation="vertical"
      class="emotion-13"
      data-dialog="true"
      data-testid="menu"
      id="id"
      role="menu"
      style="position: fixed; left: 100%; top: 100%;"
      tabindex="-1"
    >
      <button
        class="emotion-5"
        id="id-1"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        <div
          class="emotion-4"
        >
           Download
        </div>
      </button>
      <button
        class="emotion-5"
        id="id-2"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        <div
          class="emotion-4"
        >
           Link to
        </div>
      </button>
      <button
        class="emotion-5"
        id="id-3"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        <div
          class="emotion-4"
        >
           Favorite
        </div>
      </button>
      <hr
        aria-orientation="horizontal"
        class="emotion-10"
        role="separator"
      />
      <button
        class="emotion-5"
        id="id-4"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        <div
          class="emotion-4"
        >
           Delete
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
